server/extension/extension.go

```diff
@@ -300,6 +301,19 @@ type Manager struct {
 	project     ProjectGetter
 	rbac        RbacEnforcer
 	registry    ExtensionRegistry
+	metricsReg  ExtensionMetricsRegistry
+}
+
+// ExtensionMetricsRegistry exposes operations to update http metrics in the Argo CD
+// API server.
+type ExtensionMetricsRegistry interface {
+	// IncExtensionRequestCounter will increase the request counter for the given
+	// extension with the given status.
+	IncExtensionRequestCounter(extension string, status int)
+	// ObserveExtensionRequestDuration will register the request roundtrip duration
+	// between Argo CD API Server and the extension backend service for the given
+	// extension.
+	ObserveExtensionRequestDuration(extension string, duration time.Duration)
```

Comment:
That is actually a bad practice when it comes with prometheus metric. The metric cardinality would be way to high as the response byte size would vary on every request. Basically in prometheus we can't have labels with too much variation as each variation creates a new dimension in the time series.

Reply:
tks for info. 
Request duration(round trip) will also have some variation, will it cause high cardinality. Sorry if the question not relevant.

Reply:
> Sorry if the question not relevant.

Thats is ok. Valid question!

> Request duration(round trip) will also have some variation, will it cause high cardinality

It would definitely cause a super high cardinality. However, the request duration is added in a dedicated metric type called histogram. In there we define buckets and data are assigned in those buckets (see `Buckets: []float64{0.1, 0.25, .5, 1, 2, 5, 10}`). This means that regardless of how many different values we have, we are just increasing the count in the pre-defined buckets and avoid the high cardinality.